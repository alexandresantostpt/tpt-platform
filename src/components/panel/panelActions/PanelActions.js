import { travelStatus } from '@constants/travel'

import PropTypes from 'prop-types'
import React from 'react'
import styled, { css } from 'styled-components'

import { i18n } from '@i18n'

import If from '@components/if/If'

import { colors } from '@helpers/colors'
import { navigateTo } from '@utils/browser'

const checkDisabled = css`
    ${({ disabled }) => {
        if (disabled) {
            return `
                background: #dcdcdc;
                color: #dcdcdc;
                pointer-events: none;
            `
        }
    }}
`

const StyledActions = styled.ul`
    display: flex;
    list-style: none;
    margin-bottom: 3.125rem;
    margin-top: -3px;
`

const StyledAction = styled.li`
    background: ${colors.background};
    border: 0.032rem solid rgba(0, 0, 0, 0.37);
    cursor: pointer;
    ${checkDisabled};
    padding: 0.1rem 0;
    text-align: center;
    width: 100%;

    &:not(:first-child) {
        border-bottom: 0.032rem solid rgba(0, 0, 0, 0.37);
        border-left: none;
        border-top: 0.032rem solid rgba(0, 0, 0, 0.37);
    }

    &:first-child {
        border-bottom-left-radius: 0.625rem;
    }

    &:last-child {
        border-bottom-right-radius: 0.625rem;
    }
`

const Title = styled.span`
    color: ${({ theme }) => theme.text};
    font-size: 1rem;
    line-height: 1.937rem;
    opacity: 0.79;
`

const PanelActions = ({ isNewAction, onArquive, onCopy, onPreview, status, travelId }) => {
    const loadScript = () => {
        if (travelId) {
            navigateTo(`travel/${travelId}/script`)
        }
    }

    return (
        <StyledActions>
            <If condition={!!travelId}>
                <StyledAction isNewAction={isNewAction} onClick={loadScript}>
                    <Title>{i18n.t('panelActions.script')}</Title>
                </StyledAction>
            </If>
            <If condition={!!onPreview}>
                <StyledAction isNewAction={isNewAction} onClick={onPreview}>
                    <Title>{i18n.t('panelActions.preview')}</Title>
                </StyledAction>
            </If>
            <If condition={!!onCopy}>
                <StyledAction isNewAction={isNewAction} onClick={onCopy}>
                    <Title>{i18n.t('panelActions.copy')}</Title>
                </StyledAction>
            </If>
            <If condition={!!onArquive}>
                <StyledAction disabled={status !== travelStatus.CONCLUDED} isNewAction={isNewAction} onClick={onArquive}>
                    <Title>{i18n.t('panelActions.shelve')}</Title>
                </StyledAction>
            </If>
        </StyledActions>
    )
}

PanelActions.defaultProps = {
    onArquive: null,
    onCopy: null,
    onPreview: null,
    status: '',
    travelId: ''
}

PanelActions.propTypes = {
    isNewAction: PropTypes.bool.isRequired,
    onArquive: PropTypes.func,
    onCopy: PropTypes.func,
    onPreview: PropTypes.func,
    status: PropTypes.string,
    travelId: PropTypes.string
}

export default PanelActions
