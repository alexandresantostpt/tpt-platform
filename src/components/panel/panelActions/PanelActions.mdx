---
menu: Panel
name: PanelActions
route: /panel/panelActions
---
import { Playground, PropsTable } from 'docz'

import PanelActions from './PanelActions'

import '@cfg/scripts'
import '@cfg/style'

# PanelActions

<PropsTable of={ PanelActions }/>

## Basic Usage

```
<PanelActions isNewAction={ BOOLEAN_VALUE }/>
```

<Playground>
    <PanelActions isNewAction={ true }/>
</Playground>
