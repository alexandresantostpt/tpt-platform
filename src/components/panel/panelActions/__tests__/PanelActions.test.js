import React from 'react'
import renderer from 'react-test-renderer'

import PanelActions from '@components/panel/panelActions/PanelActions'

describe('Tests for PanelActions component', () => {
    it('Should rendering without crash', () => {
        const tree = renderer.create(<PanelActions isNewAction={false} />).toJSON()
        expect(tree).toMatchSnapshot()
    })
})
