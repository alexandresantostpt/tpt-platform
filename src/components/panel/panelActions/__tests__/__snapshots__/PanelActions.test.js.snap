// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for PanelActions component Should rendering without crash 1`] = `
<ul
  className="PanelActions__StyledActions-sc-8xk8c8-0 lllkGw"
/>
`;
